src/main/java/frc/robot/subsystems/elevator/ElevatorCommandBuilder.java

```diff
@@ -0,0 +1,70 @@
+package frc.robot.subsystems.elevator;
+
+import edu.wpi.first.math.controller.ElevatorFeedforward;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.FunctionalCommand;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+import frc.robot.subsystems.elevator.factories.RealElevatorConstants;
+import frc.utils.utilcommands.LoggedDashboardCommand;
+
+import java.util.function.DoubleSupplier;
+
+public class ElevatorCommandBuilder {
+
+	private final Elevator elevator;
+
+	public ElevatorCommandBuilder(Elevator elevator) {
+		this.elevator = elevator;
+	}
+
+	//@formatter:off
+    public Command setPower(double power) {
+        return new FunctionalCommand(
+                () -> {},
+                () -> elevator.setPower(power),
+                interrupted -> elevator.stop(),
+                () -> false,
+                elevator
+        ).withName("Set power: " + power);
+    }
+
+    public Command setPower(DoubleSupplier power) {
+        return new FunctionalCommand(
+                () -> {},
+                () -> elevator.setPower(power.getAsDouble()),
+                interrupted -> elevator.stop(),
+                () -> false,
+                elevator
+        ).withName("Set power: " + power);
+    }
+
+    public Command setTargetPosition(Rotation2d angle) {
+        return new InstantCommand(() -> elevator.setTargetAngle(angle));
+    }
+
+    public Command stayInPlace() {
+        return new FunctionalCommand(
+                () -> {},
+                elevator::stayInPlace,
+                interrupted -> elevator.stop(),
+                () -> false,
+                elevator
+        );
+    }
+
+    public Command stop() {
+        return new InstantCommand(elevator::stop);
+    }
+
+    public Command calibratedFeedForward() {
+        double[] oldValues = new double[] {RealElevatorConstants.FEEDFORWARD_CALCULATOR.ka, RealElevatorConstants.FEEDFORWARD_CALCULATOR.kg, RealElevatorConstants.FEEDFORWARD_CALCULATOR.kv, RealElevatorConstants.FEEDFORWARD_CALCULATOR.ks};
+        return new ParallelCommandGroup(
+                new LoggedDashboardCommand("ks", (Double newkS) -> RealElevatorConstants.FEEDFORWARD_CALCULATOR = new ElevatorFeedforward(oldValues[0], oldValues[1], oldValues[2], newkS)),
+                new LoggedDashboardCommand("kg", (Double newkG) -> RealElevatorConstants.FEEDFORWARD_CALCULATOR = new ElevatorFeedforward(oldValues[0], newkG, oldValues[2], oldValues[3]))
+        );
```

Comment:
to complicated, hust do set volt and put in code after the numbers

Reply:
wdym

Reply:
its to complected, u can just do log command that set volt to system

Reply:
so instead of changing the kG, kS in real time, calculate it and setPower? This doesn't seem to be the right method and it disables PID

Reply:
to calibrate KG, KS all u need to do is to set voltage to the motor until it stays in place and then until it moves again. No pid involves, and no reason to change constants and do control just stupid voltage will be the easiest and fastest way to calibrate it